### PR TITLE
Add executeOnJavaScriptThread methods

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

We want to add an easy way to run code on the JS thread from Expo Modules.

iOS
```
appContext?.executeOnJavaScriptThread {
  print(Thread.current.name)
}
```

Android
```
appContext.executeOnJavaScriptThread {
   System.out.println(Thread.currentThread().name)
}
```

# Test Plan

Tested manually if the code above works – on both platforms the print statements return different values inside and outside the runOnJS block.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by lukmccall — APPROVED** [review]

File: packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt [hunk 1]
```diff
@@ -320,6 +320,13 @@ class AppContext(
     return UIManagerHelper.getUIManagerForReactTag(reactContext, viewTag)?.resolveView(viewTag) as? T
   }
 
+  /**
+   * Runs a code block on the javascript thread
```
Comment (tsapeta) [inline packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:324]:
```suggestion
   * Runs a code block on the JavaScript thread.
```

File: packages/expo-modules-core/ios/Swift/AppContext.swift [hunk 2]
```diff
@@ -91,6 +91,15 @@ public final class AppContext: NSObject {
     return view as? ViewType
   }
 
+  // MARK: - Running on specific queues
+
+  /**
+   Runs a code block on the javascript thread
```
Comment (tsapeta) [inline packages/expo-modules-core/ios/Swift/AppContext.swift:97]:
```suggestion
   Runs a code block on the JavaScript thread.
```

File: packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt [hunk 3]
```diff
@@ -320,6 +320,13 @@ class AppContext(
...
+   */
+  fun runOnJS(runnable: Runnable) {
```
Comment (tsapeta) [inline packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/AppContext.kt:326]: First of all, I think it's important to add "thread" to the function name, for clarity.
What would you say about `executeOnJavaScriptThread`/`executeOnJSThread`?
We already have `runOnQueue` on the AsyncFunction component, which has much different purpose so I think it kinda makes sense to distinguish them 🤔
On the other side, we can also consider changing `runOnQueue` to something else like `setQueue` or `attachToQueue` 🤔

**Review by tsapeta — APPROVED** [review]

**Review by tsapeta — COMMENTED** [review]
One more note before you merge it: please update PR title and description to reflect applied suggestions 😉

**Pull request merged**